papaya-web/src/main/java/pl/edu/agh/papaya/rest/jira/service/JiraRestService.java

```diff
@@ -0,0 +1,185 @@
+        checkFullJiraConfig(project);
+
+        projectService.updateJiraBoardId(project, jiraBoardDto.getBoardId());
+        projectService.updateJiraProjectId(project, jiraBoardDto.getProjectId());
+    }
+
+    public List<JiraSprintDto> getAvailableJiraSprints(Sprint sprint) {
+        Project project = sprint.getProject();
+        JSONArray jiraSprints = projectService.getJiraUrl(project)
+                .flatMap(url -> jiraAuthorizationService.makeAuthorizedGetRequest(url, project.getJiraSecret(),
+                        project.getAccessToken(), String.format(JIRA_SPRINTS_ENDPOINT, project.getJiraBoardId())))
+                .orElseThrow(BadRequestException::new)
+                .getJSONObject(0)
+                .getJSONArray(VALUES);
+
+        List<JiraSprintDto> jiraSprintDtos = new LinkedList<>();
+        for (int i = 0; i < jiraSprints.length(); i++) {
+            JSONObject sprintJsonObject = jiraSprints.getJSONObject(i);
+            LocalDateTime jiraSprintStart = convertLocalDateTimeFromUtc(sprintJsonObject.getString(START_DATE));
+            if (Duration.between(jiraSprintStart, sprint.getDurationPeriod().getStart()).toMinutes()
+                    <= MAX_SPRINT_OFFSET) {
+                JiraSprintDto jiraSprintDto = new JiraSprintDto()
+                        .name(sprintJsonObject.getString(NAME))
+                        .id(sprintJsonObject.getLong(ID))
+                        .state(sprintJsonObject.getString(STATE))
+                        .startDate(jiraSprintStart);
+                jiraSprintDtos.add(jiraSprintDto);
+            }
+        }
+        return jiraSprintDtos;
+    }
+
+    private LocalDateTime convertLocalDateTimeFromUtc(String date) {
+        return LocalDateTime.parse(date, DateTimeFormatter.ISO_DATE_TIME)
+                .atZone(ZoneId.of("UTC"))
+                .withZoneSameInstant(ZoneId.systemDefault())
+                .toLocalDateTime();
+    }
+
+    public SprintStats updateFromJira(Project project, Long jiraSprintId) {
+        checkFullJiraConfig(project);
+        JSONObject jiraNumbers = projectService.getJiraUrl(project)
+                .flatMap(url -> jiraAuthorizationService.makeAuthorizedGetRequest(url, project.getJiraSecret(),
+                        project.getAccessToken(),
+                        String.format(JIRA_HOURS_ENDPOINT, jiraSprintId, project.getJiraBoardId())))
+                .orElseThrow(BadRequestException::new)
+                .getJSONObject(0)
+                .getJSONObject(CONTENTS);
+
+        SprintStats sprintStats = new SprintStats();
+        long estimatedPlanned = jiraNumbers.getJSONObject(ISSUES_NOT_COMPLETED_INITIAL_ESTIMATE_SUM).getLong(VALUE);
+        long finalPlanned = (jiraNumbers.getJSONObject(ALL_ISSUES_ESTIMATE_SUM).getLong(VALUE) + estimatedPlanned) / 2;
+        long burned = finalPlanned - jiraNumbers.getJSONObject(ISSUES_NOT_COMPLETED_ESTIMATE_SUM).getLong(VALUE);
```

Comment:
Nie, nie ufam tej matematyce - poczekam, aż front zrobi i będzie bez sensu, bo nie mam siły docierać to Postmanując

Reply:
By się jeszcze mogło aktualizować przy zamykaniu sprintu

Reply:
No i tam by można obifować sytuację, w której są same nulle, żeby też śmigało, tylko współczynnik się omijał na razie